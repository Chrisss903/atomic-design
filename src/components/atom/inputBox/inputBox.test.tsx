import { render, screen } from "@testing-library/react";
import InputBox from "./inputBox";
import "@testing-library/jest-dom";

describe("InputBox Component", () => {
  test("renders InputBox with label", () => {
    const { asFragment } = render(
      <InputBox label="Enter text" variant="outlined" children="Label" />
    );
    expect(screen.getByText("Label")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
