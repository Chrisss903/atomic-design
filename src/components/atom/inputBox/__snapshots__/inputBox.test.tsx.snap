// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputBox Component > renders InputBox with label 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-z6zit9"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Label
    </p>
    <div
      class="MuiFormControl-root MuiTextField-root css-1n40pw8-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-gpjuor"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
